tests/drivers/sensor/i3g4250d/src/main.c

```diff
@@ -0,0 +1,71 @@
+/* On device tests for I3G4250D gyro driver
+ *
+ * Copyright (c) 2021 Jonathan Hahn
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include <ztest.h>
+#include <device.h>
+#include <drivers/sensor.h>
+
+const struct device *dev;
+
+const struct device *get_sensor_devcie(void)
+{
+	const struct device *dev = device_get_binding(DT_LABEL(DT_INST(0, st_i3g4250d)));
+
+	zassert_true(device_is_ready(dev), "st i3g4250d not found");
+
+	return dev;
+}
+
+static void test_sensor_initialization(void)
+{
+	const struct device *dev = get_sensor_devcie();
+
+	/* check driver initialization */
+	zassert_not_null(dev, "failed: dev is null.");
+}
+
+static void test_get_sensor_value(void)
+{
+	const struct device *dev = get_sensor_devcie();
+	struct sensor_value val[3] = {};
+
+	/* check for fetch values */
+	int ret = sensor_sample_fetch_chan(dev, SENSOR_CHAN_ALL);
+
+	zassert_true(ret == 0, "failed fetching value");
+
+	/* check for getting values */
+	ret = sensor_channel_get(dev, SENSOR_CHAN_GYRO_XYZ, val);
+
+	zassert_true(ret == 0, "failed to get sensor value");
+}
```

Comment:
this test can likely be made generic for any gyro

Reply:
Sure, wouldn't it make sense to split the additional test into another PR?